src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectSnapshotManager.Updater.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+internal partial class ProjectSnapshotManager
+{
+    public readonly struct Updater(ProjectSnapshotManager instance)
```

Comment:
💡 it should be fairly trivial to make it so we can capture _all_ the actions done and perform them at the end. I don't think it needs to happen in this review, but that would reduce the amount of lock acquisition, threading checks, etc.

Reply:
That's a great idea for a later change! I'll file an issue.

Reply:
I think this is one potential answer to the question I was going to ask, which is simply: Whats the point of the `Updater`? I was reviewing commit-at-a-time, so was thinking maybe there would be a commit that hid all of the functionality on `ProjectSnapshotManager` so that only the `Update` and read methods were on the interface, but I didn't. Seems like the `Updater` parameter on the lambdas could be `IProjectSnapshotManager`, and the calls below pass in `this` instead of `new(this)`, and everything would work exactly the same, wouldn't it?

Reply:
#10108